### PR TITLE
Channel 7 News Script - REVISED (WIP)

### PR DESCRIPTION
**Pull request by tkcjesse**
Revised it. Much cleaner than before and has more functionality now. Will be adding more. Yes I realize it's not localized yet... 

Feel free to test it and let me know your thoughts and if you've found any issues... Will still be adding some settings stuff and then npc stuff.
 
**To test:** You can currently comment out lines **13 & 15** in core/actions/fn_newsBroadcast.sqf.
Then in debug do: [] spawn life_fnc_newsBroadcast;

@Jawshy @BoGuu Thoughts?

### CODE REVIEW

File: Altis_Life.Altis/dialog/newsStation.hpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
+class life_news_broadcast {
+    idd = 100100;
+    name= "life_news_broadcast";
+    movingEnable = 0;
+    enableSimulation = 1;
+
+    class controlsBackground {
+        class Life_RscTitleBackground: Life_RscText {
+            colorBackground[] = {"(profilenamespace getvariable ['GUI_BCG_RGB_R',0.3843])", "(profilenamespace getvariable ['GUI_BCG_RGB_G',0.7019])", "(profilenamespace getvariable ['GUI_BCG_RGB_B',0.8862])", "(profilenamespace getvariable ['GUI_BCG_RGB_A',0.7])"};
```
Comment (Jawshy) [inline Altis_Life.Altis/dialog/newsStation.hpp]: `profilenamespace`* `getvariable`*

Comment (Jawshy) [inline Altis_Life.Altis/dialog/newsStation.hpp]: Please don't hurt me.

File: Altis_Life.Altis/dialog/newsStation.hpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            x = 0.1;
+            y = 0.43;
+            w = 0.6;
+            h = .275;
+        };
+
+        class ConfirmButtonKey: Life_RscButtonMenu {
+            idc = 100104;
+            text = "$STR_News_Broadcast";
+            //onButtonClick = "[ctrlText 100101,ctrlText 100102,profilename] remoteExec ['life_fnc_AAN',-2]; closeDialog 0;";
```
Comment (Jawshy) [inline Altis_Life.Altis/dialog/newsStation.hpp]: `profileName`*

**Comment by Jawshy** [comment]
After a quick read, I see some text that could be localised. Would you like to do that or I can and then send you the changes?

**Comment by tkcjesse** [comment]
Doesn't matter, I'm just going out for a bit for some food. Just wanted people to know that this is being worked on still and not abandoned. I saw some people close some pulls yesterday and I'm wondering if that has since been abandoned now... Idk. 

If you're bored and want to you can. Otherwise I'll do it when I'm home. Some localizations in the stringtable aren't being used anymore like: `STR_News_BroadcastCost`

File: Altis_Life.Altis/Config_Master.hpp [hunk 3]
```diff
@@ -47,6 +47,11 @@ class Life_Settings {
     total_maxWeight = 24; //Static variable for the maximum weight allowed without having a backpack
     respawn_timer = 30; //How many seconds a player should wait, before being able to respawn. Minimum 5 seconds.
 
+    /* Channel 7 News Station Configurations */
+    news_broadcast_cost = 150000; //Cost for a player to send a news station broadcast.
+    news_broadcast_cooldown = 45; //Time in seconds that is required between news station broadcasts. (Default = 20 minutes)
```
Comment (Jawshy) [inline Altis_Life.Altis/Config_Master.hpp]: `1200`*

File: Altis_Life.Altis/Config_Master.hpp [hunk 4]
```diff
@@ -47,6 +47,11 @@ class Life_Settings {
...
+    news_broadcast_header_length = 1200; //Number of characters that a header can consist of. Anything over this may clip. This depends on the font size and various other factors. Adjust with caution.
```
Comment (Jawshy) [inline Altis_Life.Altis/Config_Master.hpp]: `60`*

**Comment by Jawshy** [comment]
I suggest that the `news_broadcast_cooldown` be in minutes, not seconds. Is there any reason so have it that precise? 

##### SpyGlass 
[PR #808 - Config_SpyGlass update.zip](https://github.com/ArmaLife/Framework/files/301843/PR.808.-.Config_SpyGlass.update.zip)

##### BEFilters 
[PR #808 - BEFilters update.zip](https://github.com/ArmaLife/Framework/files/301844/PR.808.-.BEFilters.update.zip)

**Comment by tkcjesse** [comment]
Absolutely. So I don't have to do any math with division or multiplication.

**Comment by TonyJohnsoniol** [comment]
Good day , we have a problem that we see the npc ATMs the channel 7 news script not could you help us ? 


this addAction[localize""STR_News_AddAction"",life_fnc_newsBroadcast,"""",1.5,false,false,"""",'isNull objectParent player && player distance _target < 4'];

**Pull request merged**